src/utils/HttpListProvider.js

```diff
@@ -0,0 +1,38 @@
+const fetch = require('node-fetch')
+
+function HttpListProvider(urls) {
+  if (!urls || !urls.length) {
+    throw new Error(`Invalid URLs: '${urls}'`)
+  }
+
+  this.urls = urls
+  this.currentIndex = 0
+}
+
+HttpListProvider.prototype.send = async function(payload, callback, retries = 0) {
+  // save the currentIndex to avoid race condition
+  const currentIndex = this.currentIndex // eslint-disable-line prefer-destructuring
+
+  if (retries === this.urls.length) {
+    callback(new Error('Request failed for all urls'))
```

Comment:
@fvictorio @akolotov what if for some reasons all of the RPC endpoints are not accessible in a short-term period, in this case, `send` method will fail here. What if to increase the level of service availability? I mean, to try to reach all RPC endpoints again. Introduce `retriesOuter` is equal, let's, say `3` - how many times `send` will try to reach all RPC endpoints. If all those tries will fail, then the method will fail. What do you think?

Reply:
Thanks @vbaranov for the comment! Actually it was requested in the original issue (https://github.com/poanetwork/bridge-nodejs/issues/28) and @fvictorio said (https://github.com/poanetwork/bridge-nodejs/pull/59#issuecomment-409891005) that is will added later after the review of main part of changes.

Reply:
oh, I see. Nevertheless, the rest looks good to me